### PR TITLE
MVD onboarding ADR

### PR DESCRIPTION
**Pull request by Izzzu**
linked to #18 

ADR describing the infrastructure needed to enable the Verifiable Credentials verification in MVD onboarding scenario.

### CODE REVIEW

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: Link to the ADR in downstream repo will be changed to the edc repo once the upstream PR is merged.

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (algattik) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: It will be used in the workflow not as a GitHub artifact. In addition, it will be published as Github artifact so that it can be used locally for deploying additional participants

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (algattik) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: In addition, a policy must be created for enrollement (require a membership VC issued by the simulated GAIA-X Authority DID)

**Review by algattik — CHANGES_REQUESTED** [review]

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (algattik) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: change to relative URL (I'm aware you're waiting for a merge, just wanted this to be tracked)

**Comment by algattik** [comment]
Added item about CLI client for the Registration Service, since it's part of the sub-flow and covers a requirement in https://github.com/agera-edc/MinimumViableDataspace/issues/45

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (zeier) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]:
```suggestion
This document outlines the infrastructure steps for the scenario of onboarding participants to the Minimum Viable Dataspace (MVD).
```

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (zeier) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]:
```suggestion
To join the MVD, each participant must have a Verifiable Credential (VC) signed by GAIA-X Authority which proves GAIA-X membership.
```

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (zeier) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: What does "configured with the simulated GAIA-X Authority DID URL" mean, and what's the purpose for that in relation to step number 7? 

Isn't the "GAIA-X Authority DID URL" already contained in the policy?

Maybe we could add an example for the policy to make it clear.

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (chrislomonico) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: @zeier , I suggested to the team that for small text changes, suggestions are committed to save back-and-forth on a PR;

**Review by chrislomonico — APPROVED** [review]
Approved pending responses to open questions

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: Yes, I already added a comment to track that

Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: https://github.com/agera-edc/MinimumViableDataspace/pull/153#discussion_r901879153

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: `The CLI client needs access to the participant's DID private key on the local file system.`
@algattik  Isn't it too detailed for this doc? We can link here the CLI documentation with instructions for local set up once available

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: I didn't include it here as it's not a part of the infrastructure, but rather an implementation feature for Registration Service

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: I updated to make it more clear. 
I also made changes in the other parts of the doc, to make the naming consistent. Please take a look again, thanks!

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: The meaning of it is that Registration Service will have some property or environment variable pointing to GAIA-X Authority DID url provided. And it's needed for the policy that verifies the VCs signed by GAIA-X Authority. I made some changes, please take a look if it's clearer now or we need more information about the policy to make it more understandable.

Comment (zeier) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: Thanks, that's clear now. I also added that the GAIA-X Authority URL will be different for each deployment and thus needs to be passed as environment variable.

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (algattik) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: We can remove "on the local file system", but the fact it needs access to the key is important I think

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (algattik) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: the thing is without this point it's not clear at all what point 5 is for. And the title is "onboarding" and not limited to infrastructure. We can discuss in PL

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (cpeeyush) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: Just a question: Do we plan to create a new separate service to simulate GAIA-X Authority?

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: Added

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: No, as far as I know we plan to only deploy DID document for GAIA-X Authority.

So I think this sentence in ADR is confusing, because we won't deploy the whole Gaia-X Authority service but only DID doc. @algattik?

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (cpeeyush) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: nit: We also have below actor named as "Authority" and Participant B is also called "Datasapce Authority". So, to avoid naming conflict can we say "_Participant B_ is the Registration Service with its `did:web` document" ?

What do you think? Thanks

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (cpeeyush) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: Publishing private key as artifact will expose the key, right ? Have you considered storing this key in key vault and from there this can be retrieved manually if needed ? 

What do you suggest ? Thanks

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (cpeeyush) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:32]: nit: We are using "will" or "will be" here in multiple places. I understand that right now we do not have all the items listed here as implemented that is why maybe we are proposing to do this in future. But at some point, these items will be implemented and then this ADR will not reflect the correct state of the MVD.

So maybe we can just list out necessary steps and not say that this will be implemented. For example:

_IdentityHub CLI client used in the MVD deployment workflow to populate the participant's Identity Hub with GAIA-X_

Comment (cpeeyush) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:32]: Suggestion can be ignored if you do not agree. Fine for me. Thanks.

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (algattik) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: Correct @Izzzu

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (algattik) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: I think it's the easiest approach, and this is not a production system. But, we don't need to delay the adoption of the ADR determining the overall design until we decide this point of detail. So we could rewrite this in a more general way and decide later. e.g. "Additionally, it will be made available to be used locally to onboard additional participants."

**Review by algattik — APPROVED** [review]

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (cpeeyush) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: okay we can write in a more genera way then as you suggested and defer the decision. Thanks.

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (cpeeyush) [inline docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:36]: Just a question: 

Didn't fully understood why "each deployment will have a different URL for the GAIA-X Authority." ? What is the purpose of this one ?

**Review by cpeeyush — APPROVED** [review]
Approved with suggestion and pending responses to open questions. Thank you.

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (zeier) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:36]: @cpeeyush The GAIA-X Authority will be implemented in the form of a DID document, which is hosted at a specific URL. Since every deployment has a different DID document (due to the keys used), every deployment needs to have it's own URL where the DID document is hosted.

Comment (cpeeyush) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:36]: @zeier okay but I was thinking GAIA-X Authority can be deployed once during very first infra setup (where we deploy ACR and storage accounts for terraform states), its keys can be stored in common key vault and this authority can be reused for every other MVD dataspace.

This will be closer to future scenario when we move to start using real Gaia-X authority service as that service is always there and not redeployed with a dataspace deployment.

But as discussed with @Izzzu this is not to hold this PR, we can merge this PR as it is and can discuss this later.

(Anyways this is just an idea as we are using a mock Gaia-X authority here)

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: I changed the  `A simulated GAIA-X Authority` to `A simulated GAIA-X Authority DID document` to make it clear that we don't deploy any services for GAIA-X

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:36]: @cpeeyush as we discussed I added an explanation in point 5 to make clear that the approach of deploying the GAIA-X Authority every time with the Dataspace is a simplification, but in general it's an external central component:

`For the simplicity the GAIA-X Authority DID document 
   is deployed together with other Dataspace components, which means that each deployed Dataspace will have its own GAIA-X Authority instance. In a
   real scenario the GAIA-X Authority is an external component and can communicate with multiple Dataspaces.`
   
   Please take a look :)

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: good point, I made it more general as you suggested. The details can be a part of other documentation created during the implementation

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (zeier) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:36]: @cpeeyush that's a great catch! I think both approaches work, while a single GAIA-X authority is closer to reality. let's decide on one approach and then move on.

**Review by zeier — APPROVED** [review]

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: agree, changed

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md:32]: As we discussed - I will leave this for now as we have also the developers records dates in the folder names so we can see in this case that this doc presents the plan for the future. Therefore we have "will" and "will be" but agree on the point that we can make the wording more neutral in terms of the used time as some of the components mentioned already exist. Thanks

**Review by ouphi — APPROVED** [review]

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: done

File: docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md
Comment (Izzzu) [reply docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md]: changed

**Pull request merged**
